bftengine/src/bftengine/KeyStore.h

```diff
@@ -98,4 +100,28 @@ class ClusterKeyStore : public ResPagesClient<ClusterKeyStore> {
 
   std::string buffer_;
 };
+
+// Manages the BFT state of the clients public keys:
+// It stores the digest of the keys in the reserved pages and sets the published_ accordingly
+class ClientKeyStore : public ResPagesClient<ClientKeyStore, 1> {
+ public:
+  bool published_{false};
```

Comment:
I guess `published_` variable will be removed when we'll load actual keys rather than a hash, then we will be able to check a key container size to understand whether keys are published.

Reply:
but then you'll do it each time you query whether keys were published?